local/local.go

```diff
@@ -351,11 +354,15 @@ func (i *Image) doSave() (types.ImageInspect, error) {
 			done <- err
 			return
 		}
+
+		if err := checkResponseError(res.Body); err != nil {
+			done <- err // add error but continue, to ensure reader is closed
+		}
+
 		if err := ensureReaderClosed(res.Body); err != nil {
 			done <- errors.Wrap(err, "failed to drain and close response from docker client")
 			return
 		}
```

Comment:
I rebased against the changes from #28, and I hope this is the now- intended behavior - adding any `res.Body` errors to the channel but deferring return until `ensureReaderClosed` can flush and close the rest of the response. Look ok?

Reply:
Sending on the `done` chan will unblock the next interaction with the docker client (`i.docker.ImageInspectWithRaw` on line 423). We need to flush and close the response before sending another request to the daemon. We could save the error to return after we ensureReaderClose (instead of the close error if there is one).

Reply:
Oh cool. I'll try that then! Thanks!